htdocs/core/lib/images.lib.php

```diff
@@ -306,35 +306,94 @@ function dol_imageResizeOrCrop($file, $mode, $newWidth, $newHeight, $src_x = 0,
  */
 function dolRotateImage($file_path)
 {
-    $exif = @exif_read_data($file_path);
-    if ($exif === false) {
-        return false;
-    }
-    $orientation = intval(@$exif['Orientation']);
-    if (!in_array($orientation, array(3, 6, 8))) {
-        return false;
-    }
-    $image = @imagecreatefromjpeg($file_path);
-    switch ($orientation) {
-        case 3:
-            $image = @imagerotate($image, 180, 0);
-            break;
-        case 6:
-            $image = @imagerotate($image, 270, 0);
-            break;
-        case 8:
-            $image = @imagerotate($image, 90, 0);
-            break;
-        default:
-            return false;
-    }
-    $success = imagejpeg($image, $file_path);
-    // Free up memory (imagedestroy does not delete files):
-    @imagedestroy($image);
-    return $success;
+	return correctExifImageOrientation($file_path, $file_path);
 }
 
 
+/**
+ * Add exif orientation correction for image
+ *
+ * @param $fileSource string
```

Comment:
@atm-john `@param string $fileSource more explanation if needed`

Reply:
Oups...
thank you !